embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/SpansRepository.kt

```diff
@@ -0,0 +1,62 @@
+package io.embrace.android.embracesdk.internal.spans
+
+import io.embrace.android.embracesdk.spans.EmbraceSpan
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Allows the tracking of [EmbraceSpan] instances so that their references can be retrieved with its associated spanId
+ */
+internal class SpansRepository {
+    private val activeSpans: MutableMap<String, EmbraceSpan> = ConcurrentHashMap()
```

Comment:
Strictly speaking I don't think this needs to be `ConcurrentHashMap` as access is always synchronised, it's fine to leave it as-is though

Reply:
I'll change that. No need to pay the penalty if we don't have to.